packages/ra-ui-materialui/src/input/NumberInput.tsx

```diff
@@ -87,7 +89,7 @@ NumberInput.propTypes = {
     options: PropTypes.object,
     resource: PropTypes.string,
     source: PropTypes.string,
-    step: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
+    step: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
```

Comment:
I might be wrong, but `step` is not in `InputProps<TextFieldProps>` nor in `Omit<TextFieldProps, 'label' | 'helperText'>`.

Should we add this prop in a new interface or something?